pt_miniscreen/pages/system/memory.py

```diff
@@ -0,0 +1,87 @@
+import psutil
+from pitop.common.formatting import bytes2human
+
+from ...hotspots.base import HotspotInstance
+from ...hotspots.progress_bar import Hotspot as ProgressBarHotspot
+from ...hotspots.templates.marquee_dynamic_text import (
+    Hotspot as MarqueeDynamicTextHotspot,
+)
+from ...hotspots.templates.text import Hotspot as TextHotspot
+from ..base import Page as PageBase
+
+X_MARGIN = 5
+SUB_TITLE_WIDTH = 40
+ROW_HEIGHT = 10
+TITLE_FONT_SIZE = 12
+TEXT_FONT_SIZE = 10
+MARGIN_Y = 5
+SPACING_Y = 2
+
+
+class Page(PageBase):
+    def __init__(self, size):
+        super().__init__(size=size)
+        self.reset()
+
+    def reset(self):
+        try:
+            memory = psutil.virtual_memory()
```

Comment:
Since this is calculated once and not updated the memory doesn't change until reset is called again. I think `psutil.virtual_memory` and `psutil.swap_memory` need to be moved into the functions that are passed into the marquee text and progress bar hotspots.